core/frontend/src/tile/map/ArcGisTokenGenerator.ts

```diff
@@ -0,0 +1,169 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { request, RequestOptions } from "@bentley/itwin-client";
+import { FrontendRequestContext } from "../../imodeljs-frontend";
+
+/** @packageDocumentation
+ * @module Tiles
+ */
+
+/** @internal */
+export interface ArcGisToken {
+  // The generated token.
+  token: string;
+
+  // The expiration time of the token in milliseconds since January 1, 1970 (UTC).
+  expires: number;
+
+  // This property will show as true if the token must always pass over ssl.
+  ssl: boolean;
+}
+
+// client application's base URL, a user-specified IP address, or the IP address that is making the request.
+/** @internal */
+export enum ArcGisTokenClientType {
+  ip,
+  referer,
+  requestIp,
+}
+
+/** @internal */
+export interface ArcGisGenerateTokenOptions {
+
+  // The client type that will be granted access to the token.
+  // Users will be able to specify whether the token will be generated for a client application's base URL,
+  // a user-specified IP address, or the IP address that is making the request.
+  client: ArcGisTokenClientType;
+
+  // The IP address that will be using the created token for access.
+  // On the Generate Token page, the IP address is specified in the IP Address field. This is required when client has been set as ip.
+  // Example ip=11.11.111.111
```

Comment:
`11.11.111.111` is not a `number` - how does this work?

Reply:
good catch, that should be a string.